spec/helpers/faker/puppet_module_remote_repo.rb

```diff
@@ -0,0 +1,140 @@
+require 'open3'
+
+require_relative '../faker'
+
+module ModuleSync
+  # Fake a remote git repository that holds a puppet module
+  #
+  # This module allows to fake a remote repositiory using:
+  #  - a bare repo
+  #  - a temporary cloned repo to operate on the remote before exposing to modulesync
+  #
+  # Note: This module needs to have working_directory sets before using it
+  module Faker
+    class PuppetModuleRemoteRepo
+      class CommandExecutionError < StandardError; end
+
+      attr_reader :name
+      attr_reader :namespace
+
+      def initialize(name, namespace)
+        @name = name
+        @namespace = namespace
+      end
+
+      def populate
+        FileUtils.chdir(Faker.working_directory) do
+          run %W{git init --bare #{bare_repo_dir}}
+          run %W{git clone #{bare_repo_dir} #{tmp_repo_dir}}
+
+          module_short_name = name.split('-').last
+
+          FileUtils.chdir(tmp_repo_dir) do
+            metadata = {
+              name: "modulesync-#{module_short_name}",
+              version: '0.4.2',
+              author: 'ModuleSync team',
+            }
+
+            File.write 'metadata.json', metadata.to_json
+            run %w{git add metadata.json}
+            run %W{git commit --message} << 'Initial commit'
+            run %w{git push}
+          end
+        end
+      end
+
+      def read_only=(value)
+        mode = value ? '0444' : '0644'
+        FileUtils.chdir(bare_repo_dir) do
+          run %W{git config core.sharedRepository #{mode}}
+        end
+      end
+
+      def default_branch
+        FileUtils.chdir(bare_repo_dir) do
+          stdout = run %w{git symbolic-ref --short HEAD}
+          return stdout.chomp
+        end
+      end
+
+      def default_branch=(value)
+        FileUtils.chdir(bare_repo_dir) do
+          run %W{git branch -M #{default_branch} #{value}}
+          run %W{git symbolic-ref HEAD refs/heads/#{value}}
+        end
+      end
+
+      def read_file(filename, branch = nil)
+        branch ||= default_branch
+        FileUtils.chdir(bare_repo_dir) do
+          return run %W{git show #{branch}:#{filename}}
+        rescue CommandExecutionError
+          return nil
```

Comment:
I think this and the other return are redundant. In Ruby the last statement is always a return. I'm fine either way, but perhaps Rubocop will yell at you.

Reply:
`FileUtils.chdir` relies en `Dir.cd`, which returns the block result, so, yes, the returns are redundant, but as the concerned code is in a block, I seems to me more understandable to return explicitly.
FYI, Rubocop did tell me anything about this, even using a recent version (ie. `1.7.0`, while `modulesync` uses `0.50`).

Reply:
I agree return here is more consistent so :+1: to keeping it this way.